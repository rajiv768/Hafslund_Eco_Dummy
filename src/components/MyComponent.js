/* eslint-disable no-console */
import React, { useEffect, useState, useContext } from 'react';
import PropTypes from 'prop-types';
import { Header } from 'semantic-ui-react';

import { WidgetContext } from './Context';
import MyChildComponent from './MyChildComponent';

const MyComponent = ({ title }) => {
  const context = useContext(WidgetContext);
  // context object will contain:
  /*
    {
      wab,
      esriJS,
      onOpen,
      onClose,
      loading,
      error,
      layers,
      version
    }
  */
  // console.log('Widget Context:', context);

  // set state
  const [msg, setMsg] = useState('WAB');
  const [open, setOpen] = useState(true);

  // set message
  const updateMsg = newMsg => {
    setTimeout(() => {
      setMsg(newMsg);
    }, 1000);
  };

  // handle widget open and close
  const handleOpen = () => {
    console.log('Open widget');
    setOpen(true);
  };

  const handleClose = () => {
    console.log('Close widget');
    setOpen(false);
  };

  // lifecycle
  useEffect(() => {
    // init open/close handlers
    context.onOpen(handleOpen);
    context.onClose(handleClose);
    // set msg
    updateMsg('WAB React Widget Template');
  }, []);

  if (context.loading) {
    return <div>Loading...</div>;
  }

  if (context.error) {
    return <div>Error...</div>;
  }

  return (
    <div css="padding: 0.25rem;">
      <Header>{title}</Header>
      <MyChildComponent msg={msg} open={open} />
      <div css="font-size: 0.7rem; color: #999999ab;">
        Version: {context.version}
      </div>
    </div>
  );
};

MyComponent.propTypes = {
  title: PropTypes.string,
};

MyComponent.defaultProps = {
  title: 'Hello',
};

export default MyComponent;
