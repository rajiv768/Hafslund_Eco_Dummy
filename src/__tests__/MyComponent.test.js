import React from 'react';
import '@testing-library/react/cleanup-after-each';
import { render, cleanup, wait } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import MyComponent from '../components/MyComponent';

afterEach(cleanup);

// https://testing-library.com/docs/example-react-context
describe('<MyComponent />', () => {
  it('renders main component without error', async () => {
    const { container, getByText } = render(<MyComponent />);
    expect(container.firstChild).toMatchSnapshot();
    expect(getByText('WAB')).toBeInTheDocument();

    // wait for useEffect
    await wait(() =>
      expect(getByText('WAB React Widget Template')).toBeInTheDocument(),
    );
  });
});
