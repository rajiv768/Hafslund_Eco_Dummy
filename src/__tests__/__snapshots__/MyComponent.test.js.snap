// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyComponent /> renders main component without error 1`] = `
<div
  class="sc-bdVaJa jBIjIO"
>
  <div
    class="ui header"
  >
    Hello
  </div>
  <div
    class="ui segment"
  >
    <h3>
      WAB
    </h3>
    <h5>
      Widget Open
    </h5>
  </div>
  <div
    class="sc-bwzfXH fFwXog"
  >
    Version: 
  </div>
</div>
`;
